src/middleware/cache/index.ts

```diff
@@ -5,28 +5,60 @@ export const cache = (options: {
   cacheName: string
   wait?: boolean
   cacheControl?: string
+  vary?: string
 }): MiddlewareHandler => {
   if (options.wait === undefined) {
     options.wait = false
   }
 
-  const directives = options.cacheControl?.split(',').map((directive) => directive.toLowerCase())
+  const cacheControlDirectives = options.cacheControl
+    ?.split(',')
+    .map((directive) => directive.toLowerCase())
+  const varyDirectives = options.vary?.split(',').map((directive) => directive.trim())
+  // RFC 7231 Section 7.1.4 specifies that "*" is not allowed in Vary header.
+  // See: https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.4
+  if (varyDirectives?.includes('*')) {
+    throw new Error(
+      'Middleware vary configuration cannot include "*", as it disallows effective caching.'
+    )
+  }
 
   const addHeader = (c: Context) => {
-    if (directives) {
+    if (cacheControlDirectives) {
       const existingDirectives =
         c.res.headers
           .get('Cache-Control')
           ?.split(',')
           .map((d) => d.trim().split('=', 1)[0]) ?? []
-      for (const directive of directives) {
+      for (const directive of cacheControlDirectives) {
         let [name, value] = directive.trim().split('=', 2)
         name = name.toLowerCase()
         if (!existingDirectives.includes(name)) {
           c.header('Cache-Control', `${name}${value ? `=${value}` : ''}`, { append: true })
         }
       }
     }
+
+    if (varyDirectives) {
+      const existingDirectives =
+        c.res.headers
+          .get('Vary')
+          ?.split(',')
+          .map((d) => d.trim()) ?? []
+      const vary = Array.from(
+        new Set(
+          [...existingDirectives, ...varyDirectives].map((directive) =>
+            directive.toLocaleLowerCase()
+          )
+        )
+      ).sort((a, b) => a.localeCompare(b))
```

Comment:
I sorted the `Vary` to be given to the header, but I am willing to remove it if it affects performance.

Reply:
This is a matter for discussion, but I think sorting is good. Because there are cases where the cache keys are different if the header values are in a different order.